src/Components/SystemDetail/SystemRecommendations.js

```diff
@@ -24,26 +25,89 @@ const RecommendationsTable = asyncComponent(() => import('./RecommendationsTable
  *
  * @param props the props given by the smart component.
  */
+const defaultFilters = {
+    name: {
+        label: 'Name',
+        value: ''
+    }
+};
+
+export const isEmptyFilters = (activeFilters) => Object.values(activeFilters).find(
+    (item) => item?.value?.length > 0 || item?.length > 0
+);
+
+export const constructActiveFilters = (activeFilters) => Object.entries(activeFilters).map(([key, { label, value } = {}]) => ({
+    category: label,
+    chipKey: key,
+    chips:
+        value?.length > 0
+            ? Array.isArray(value)
+                ? value.map((item) => ({ name: item })) : [{ name: value }]
+            : []
+}));
+
+export const onDeleteFilter = (activeFilters, itemsToRemove) => {
+    const currItem = itemsToRemove[0];
+    return {
+        ...activeFilters,
+        [currItem?.chipKey]: {
+            ...(activeFilters[currItem?.chipKey] || {}),
+            value: Array.isArray(activeFilters[currItem?.chipKey]?.value)
+                ? activeFilters[currItem?.chipKey]?.value?.filter(
+                    (item) => !currItem?.chips?.find(({ name }) => name === item)
+                  )
+                : ''
+        }
+    };
+};
 
 class SystemRecommendations extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
             page: 1,
             perPage: 10,
-            inventoryId: props.match.params.inventoryId
+            inventoryId: props.match.params.inventoryId,
+            activeFilters: defaultFilters
         };
+
+        this.throttleHandleChange = debounce(this.throttleHandleChange.bind(this), 500);
+
     }
 
     async componentDidMount() {
-        await this.props.loadSysRecs(this.state.inventoryId,
-            { page: this.state.page, perPage: this.state.perPage }
-        );
+        await this.requestLoadSysRecs();
+    }
+
+    async throttleHandleChange(options) {
+        await this.requestLoadSysRecs(options);
+    }
+
+    async requestLoadSysRecs(options = {}) {
+        const params = {
+            page: this.state.page,
+            perPage: this.state.perPage,
+            description: this.state?.descriptionFilter,
+            ...options
+        };
+        await this.props.loadSysRecs(this.state.inventoryId, params);
+    }
+
+    applyFilters(filters) {
+        this.setState({
+            activeFilters: filters,
+            descriptionFilter: filters?.name?.value,
+            page: 1
+        });
+        this.throttleHandleChange({
```

Comment:
You'd use `debouncedThrottleHandleChange` in here, instead of `throttleHandleChange`
```suggestion
        this.debouncedThrottleHandleChange({
```